### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by philj404**
Uses the <AUnit.h> library to collect and run unit tests on the Arduino platform.

_Users_ of <arduino-timer.h> do not depend on AUnit.  Just _developers_ that want to run those particular test script(s) will need to add AUnit to their IDE.

example output:
```
Running C:\Users\philk\OneDrive\Documents\Arduino\arduino-timer\extras\tests\testTimer\testTimer.ino, Built Nov  2 2020
TestRunner started on 5 test(s).
Test timer_at passed.
Test timer_cancelTasks passed.
Test timer_delayToNextEvent passed.
Test timer_every passed.
Test timer_in passed.
TestRunner duration: 0.040 seconds.
TestRunner summary: 5 passed, 0 failed, 0 skipped, 0 timed out, out of 5 test(s).

```

### CODE REVIEW

File: src/arduino-timer.h [hunk 1]
```diff
@@ -96,6 +96,16 @@ class Timer {
         task = (Task)NULL;
     }
 
+    /* Cancel ALL timer tasks -- useful for testing */
```
Comment (philj404) [inline src/arduino-timer.h:99]: I had to reuse the same timer for multiple tests.  

I added a Timer<>::cancelAll() method so I could delete tasks where the task ID was forgotten.  I hate adding a new public interface for this, but I couldn't think of a workaround.

(I wanted to make a local test timer (a local variable in the test function), but Arduinos don't have enough stack space for that AND the test infrastructure, etc.)

If this change in public interface is unacceptable, reject the pull request.

**Comment by philj404** [comment]
See also https://github.com/contrem/arduino-timer/issues/20

File: README.md [hunk 2]
```diff
@@ -56,6 +56,11 @@ auto task = timer.in(delay, function_to_call);
 timer.cancel(task);
 ```
 
+To **cancel all pending tasks**
+```cpp
+timer.cancelAll();
```
Comment (philj404) [inline README.md:61]: New public interface.  Sorry... but it should be easy to support.

**Comment by contrem** [comment]
I implemented a cancel all version of **cancel** in [mc/cancel-all](https://github.com/contrem/arduino-timer/tree/mc/cancel-all)  because there are some pending changes affecting it. A preview is available in [mc/foreach](https://github.com/contrem/arduino-timer/tree/mc/foreach).

**Comment by philj404** [comment]
Looks good.  I will pull your changes into my branch and modify my code to use your **timer.cancel()**.

**Comment by philj404** [comment]
I just noticed you now have a [CONTRIBUTING.md](https://github.com/contrem/arduino-timer/blob/master/CONTRIBUTING.md) guideline.  Should I change my branch name (for example create a new branch pj/unitTests) or point my pull request to testing?  

Please let me know if I need to make a change to make your code merge easier.

**Comment by contrem** [comment]
> Should I change my branch name (for example create a new branch pj/unitTests)

I'll handle that on my end.

> or point my pull request to testing?

Nope, changes should pretty much always be aimed at master.

Also since I added the commit / styling guidelines after you already made this pull request, and since it's not to the main code base, it'd be unfair to force them upon you now.

File: keywords.txt [hunk 3]
```diff
@@ -20,6 +20,7 @@ every	KEYWORD2
 tick	KEYWORD2
 ticks	KEYWORD2
 cancel	KEYWORD2
+cancelAll	KEYWORD2
```
Comment (contrem) [inline keywords.txt:23]: No longer required

File: extras/tests/testTimer/testTimer.ino [hunk 4]
```diff
@@ -0,0 +1,314 @@
+
+//not needed: #line 2 "testTimer.ino"
+//
```
Comment (contrem) [inline extras/tests/testTimer/testTimer.ino:3]: Use `git diff --check` to help find trailing whitespace:

```
extras/tests/testTimer/testTimer.ino:3: trailing whitespace.
+// 
extras/tests/testTimer/testTimer.ino:15: trailing whitespace.
+// forcing tests to slow down. 
```

Comment (philj404) [reply extras/tests/testTimer/testTimer.ino]: I didn't know about "git diff --check.  And my version of "git diff" doesn't seem to find the whitespace.  I'll check for an update. 

Thanks for pointing this out.

**Comment by philj404** [comment]
This pull request is an early subset of #21 .  *That* PR will merge it all in.

**Pull request closed without merging**